pkg/nsx/services/subnetbinding/subnetbinding.go

```diff
@@ -0,0 +1,235 @@
+	if len(bindingMapsByChild) > 0 {
+		return bindingMapsByChild
+	}
+	return s.GetSubnetConnectionBindingMapsByParentSubnet(subnet)
+}
+
+// GetSubnetConnectionBindingMapsByChildSubnet returns the SubnetConnectionBindingMaps associated with the subnet.
+func (s *BindingService) GetSubnetConnectionBindingMapsByChildSubnet(subnet *model.VpcSubnet) []*model.SubnetConnectionBindingMap {
+	return s.BindingStore.getBindingsByChildSubnet(*subnet.Path)
+}
+
+// GetSubnetConnectionBindingMapsByParentSubnet returns the SubnetConnectionBindingMaps connected to the subnet.
+func (s *BindingService) GetSubnetConnectionBindingMapsByParentSubnet(subnet *model.VpcSubnet) []*model.SubnetConnectionBindingMap {
+	return s.BindingStore.getBindingsByParentSubnet(*subnet.Path)
+}
+
+func (s *BindingService) GetSubnetConnectionBindingMapCRsBySubnet(subnet *model.VpcSubnet) []*v1alpha1.SubnetConnectionBindingMap {
+	nsxBindingMaps := s.GetSubnetConnectionBindingMapsBySubnet(subnet)
+	if len(nsxBindingMaps) == 0 {
+		return nil
+	}
+
+	subnetConnectionBindingMaps := sets.New[*v1alpha1.SubnetConnectionBindingMap]()
+	for _, bm := range nsxBindingMaps {
+		obj, err := buildSubnetConnectionBindingMapCR(bm)
+		if err != nil {
+			log.Error(err, "Unable to get SubnetConnectionBindingMap CR from NSX resource", "SubnetConnectionBindingMap", *bm.Id)
+			continue
+		}
+		subnetConnectionBindingMaps.Insert(obj)
+	}
+	return subnetConnectionBindingMaps.UnsortedList()
+}
+
+func (s *BindingService) ListSubnetConnectionBindingMapCRUIDsInStore() sets.Set[string] {
+	crUIDs := sets.New[string]()
+	for _, obj := range s.BindingStore.List() {
+		bm, _ := obj.(*model.SubnetConnectionBindingMap)
+		res, err := buildSubnetConnectionBindingMapCR(bm)
+		if err != nil {
+			log.Error(err, "Unable to get SubnetConnectionBindingMap CR from NSX resource", "bindingMap", *bm.Id)
+			continue
+		}
+		crUIDs.Insert(string(res.UID))
+	}
+	return crUIDs
+}
+
+// Apply sync bindingMaps on NSX and save into the store if succeeded to realize.
+func (s *BindingService) Apply(subnetPath string, bindingMaps []*model.SubnetConnectionBindingMap) error {
+	return s.hUpdateSubnetConnectionBindingMaps(subnetPath, bindingMaps)
+}
+
+// deleteSubnetConnectionBindingMaps uses HAPI call to delete multiple SubnetConnectionBindingMaps on NSX in one
+// transaction, and then delete these resources from local store. Since NSX can accept no more than 5K children Nodes
+// in one HAPI call, paging is used in this function to avoid scale issues.
+func (s *BindingService) deleteSubnetConnectionBindingMaps(bindingMaps []*model.SubnetConnectionBindingMap) error {
+	// Add this check for security purpose. The caller has similar pre-check and returned if no items exist in the bindingMaps.
+	bindingMapsCount := len(bindingMaps)
+	if bindingMapsCount == 0 {
+		log.Info("No existing SubnetConnectionBindingMaps found in the store")
+		return nil
+	}
+	pages := (bindingMapsCount + hAPIPageSize - 1) / hAPIPageSize
```

Comment:
Can you consider adding a unit test for the paging logic. Just to ensure the pages calculation and loop are correct.

Reply:
We can do this in a follow-up patch.

Reply:
Sure, I will add the UT in the follow-up change.